determined-ai/determined
# fix: add jobIDToRunningPods to k8sResourcePool for accurate slot count

## Description
[DET-8759]
Add a jobIDToRunningPods field to kubernetesResourcePool to accurately increment & count slots used per jobID. 
This fixes inaccurate slot counts for deadlocked jobs.

## Test Plan
To test this change:
- Spin up a GKE cluster with 2 nodes. (Resources: [Setup](https://docs.determined.ai/latest/cluster-setup-guide/deploy-cluster/sysadmin-deploy-on-k8s/setup-gke-cluster.html), [Install Determined](https://docs.determined.ai/latest/cluster-setup-guide/deploy-cluster/sysadmin-deploy-on-k8s/install-on-kubernetes.html#install-on-kubernetes))

- Start several Determined experiments with --config=resources.slots_per_trial=2. 
`det e create <config file> <context directory>--config="resources.slots_per_trial=2"`

- Eventually, the experiments will deadlock and should appear as follows:
`   # | ID                                   | Type            | Job Name                    |   Weight | Submitted                 | Slots (acquired/needed)   | State           | User`
`  0 | 9f32a1fe-6e9e-4902-b090-70b4d2cfda77 | TYPE_EXPERIMENT | mnist_pytorch_const (16) |        1 | 2022-12-12 19:54:42+00:00 | 1/2                       | STATE_SCHEDULED | determined `
`    1 | 1708336e-9021-42a9-acbd-4a156f21f14d | TYPE_EXPERIMENT | mnist_pytorch_const (17) |        1 | 2022-12-12 19:54:56+00:00 | 1/2                       | STATE_SCHEDULED | determined `
`      2 | c0b598e3-7605-4d42-8a0d-37b3323227cd | TYPE_EXPERIMENT | mnist_pytorch_const (18) |        1 | 2022-12-12 19:54:58+00:00 | 0/2                       | STATE_QUEUED    | determined `
`        3 | e45d6792-e726-4404-a2aa-78b14c9d0401 | TYPE_EXPERIMENT | mnist_pytorch_const (19) |        1 | 2022-12-12 19:54:59+00:00 | 0/2                       | STATE_QUEUED    | determined`
  Note that the scheduled experiments show 1/2 slots _not_ 2/2.

## Ticket
DET-8759

[DET-8759]: https://hpe-aiatscale.atlassian.net/browse/DET-8759?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
